internal/infrastructure/kubernetes/resource/resource.go

```diff
@@ -31,3 +33,9 @@ func ExpectedServiceSpec(serviceType *egcfgv1a1.ServiceType) corev1.ServiceSpec
 	}
 	return serviceSpec
 }
+
+// CompareSvc Only compare the selector and ports(not include nodePort) in case user have modified for some scene.
+func CompareSvc(currentSvc, originalSvc *corev1.Service) bool {
+	return cmp.Equal(currentSvc.Spec.Selector, originalSvc.Spec.Selector) &&
```

Comment:
what about service Type, since we support configuring it, should we also ensure it is not being changed

Reply:
yes, i'll made svc'type compare if my [question](https://github.com/envoyproxy/gateway/issues/1373) solved.

Reply:
#1373 I will fix it ASAP